### DISCUSSION TITLE
What is configuration to connect to sql server localdb -> (LocalDB)\MSSQLLocalDB

### DISCUSSION THREAD

**Discussion by omerkocadere** [dbeaver/dbeaver#14947] [Q&A]
I could not achieve to connect. Could you please show how to connect?

It is working when trying to connect using sql server
![Ssms_KYiJDjFd9o](https://user-images.githubusercontent.com/9765632/146677919-53d0ebcc-481c-442c-857e-a0a46281b146.png)
?

But gives an error when trying with dbeaver
![dbeaver_JjbE6OKBzQ](https://user-images.githubusercontent.com/9765632/146677989-93a46263-5e60-44da-a116-94aa0ac93cdb.png)
![dbeaver_i4KnXSJ9Am](https://user-images.githubusercontent.com/9765632/146678002-1786f055-c6c9-4b46-96fb-bf34b547ae10.png)

**Comment by ShadelessFox** [comment]
Hello @zormaxx,

You should specify `localhost` as a hostname.

**Reply by omerkocadere** [reply]
Did not connect to localhost? Please see my new screenshot.

**Comment by omerkocadere** [comment]
Did not connect to localhost? Do you know other ways?

![image](https://user-images.githubusercontent.com/9765632/146735453-92cd82ac-0dcf-474c-ae46-31543fda9f1c.png)

**Reply by ShadelessFox** [reply]
It looks like that your SQL Server service is not running. Is it possible to connect to the same database using SSMS?

**Reply by omerkocadere** [reply]
I can connect using SSMS. Actually, I didn't install Sql Server Express externally, localdb is installed by Visual Studio 2022 while installing it.

![0K1yZuiKBh](https://user-images.githubusercontent.com/9765632/146754602-f8334866-dcde-4123-8b04-e4bf49bd1fc9.gif)

**Reply by ShadelessFox** [reply]
Can you manually check whether the service is running?

**Comment by rambouillet** [comment]
@zormaxo @ShadelessFox 

I found a way to get DBeaver connected to localdb using ODBC, which persists across the localdb pipe name changes. You only need to have an `ODBC Driver -- for SQL Server` installed. Check it out:

1. First verify you have a SQL Server ODBC driver installed. To do this, go to the Windows Control Panel >> Administrative Tools >> ODBC Data Sources (x64) >> "Drivers" Tab. The driver I have on my machine is `ODBC Driver 17 for SQL Server`. If you don't have the driver, you can download it here: https://docs.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver15

The driver will look like this:

![odbc-driver](https://user-images.githubusercontent.com/20598632/161351237-2497f2ba-eb0c-45ec-ae76-f0799888c525.PNG)

2. Once you have the ODBC driver, go to DBeaver and create a new connection by choosing the "ODBC" option, and click Next:

![odbc-conn1](https://user-images.githubusercontent.com/20598632/161351516-94711fa6-5ef5-4479-8a47-ee4283ea0798.PNG)

3. Enter in the connection string as follows. Be sure to change the ODBC driver version number to the one that you have installed, as well as the localdb instance name:

`Driver={ODBC Driver 17 for SQL Server};Server=(localdb)\mssqllocaldb`

![odbc-conn2](https://user-images.githubusercontent.com/20598632/161351860-7bf6c2cf-4b1f-4670-91db-c3dd67fb91c9.PNG)

4. Click "Test Connection" and see if it works. Hopefully you don't have any problems. Let me know if this works well for you.

![dbeaver](https://user-images.githubusercontent.com/20598632/161352167-8882033d-ff32-4e36-8ee5-81b27f4f757b.PNG)

**Reply by ShadelessFox** [reply]
The ODBC driver lacks a lot of features provided by the specialized SQL Server driver.

You should not use it until there's no JDBC driver for the target database.

**Reply by rambouillet** [reply]
Oh darn, what features are missing in the ODBC version? Is the ODBC version at least good enough for just viewing the data?

**Reply by ShadelessFox** [reply]
It lacks driver-specific support for handling data and metadata since it's a generic driver. Also, ODBC has pretty limited API when comparing to JDBC, e.g. no support for procedures.

It's OK for viewing the data, but I'd recommend sticking to the SQL Server driver.

**Reply by alexandredsimoes** [reply]
![image](https://user-images.githubusercontent.com/5480346/165954447-e72f4d11-5d97-43ba-8e70-75b587c75e18.png)
Now it connects but shows an error when I try to list tables.

**Comment by smourier** [comment]
IMHO, the root cause is the SQL Server driver doesn't currently support this: https://github.com/microsoft/mssql-jdbc/issues/769
